### DISCUSSION TITLE
Fix for grpc crashing in recent Ansible versions

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#875] [General]
All,

There are some breaking changes in the plugin framework for Ansible, which cause issues with the grpc plugin used for SR Linux and SR OS.

I have a fix available for quite some time: https://github.com/jbemmel/ansible-networking-collections/blob/master/grpc/plugins/connection/gnmi.py#L268

Unfortunately, I have not been able to get the upstream plugin updated with this small 1-line change.

It would be possible to modify the grpc install script to do:
```ansible-galaxy collection install 'git+https://github.com/jbemmel/ansible-networking-collections.git#/grpc/'```

I think that may be a better solution than downgrading Ansible - although I'm not too keen on taking responsibility for maintaining that plugin.

Just wanted to put this out there, up to the team to decide what we prefer

**Comment by ipspace** [comment]
> I think that may be a better solution than downgrading Ansible - although I'm not too keen on taking responsibility for maintaining that plugin.

I have no problems with downgrading Ansible. After all, so far we're using only baseline functionality that's been around for ages.

**Comment by ipspace** [comment]
@jbemmel Any chance of getting a fixed plugin or shall I close this discussion?

**Reply by jbemmel** [reply]
I am working on fixing this issue upstream: https://github.com/ansible/ansible/pull/82475#issuecomment-1876089954
(Nokia GRPC plugin isn't the only component impacted by this)

**Comment by ipspace** [comment]
When it's impossible to persuade the prophet to come to the mountain, the mountain is moved ;) Ansible 9.5.1 includes a fix that makes it work with the ancient Nokia GRPC collection.
